build.py

```diff
@@ -417,39 +421,47 @@ def main():
     use_cache = ARGS.use_cache and os.path.isfile(cache_path)
-    with open(os.path.join(ARGS.model_path, "config.json"), encoding="utf-8") as i_f:
-        config = json.load(i_f)
-        if not use_cache:
-            if ARGS.model_category == "llama":
-                mod, params = llama.get_model(ARGS, config)
-            elif ARGS.model_category == "gpt_neox":
-                mod, params = gpt_neox.get_model(ARGS, config)
-            elif ARGS.model_category == "gpt_bigcode":
-                mod, params = gpt_bigcode.get_model(ARGS, config)
-            elif ARGS.model_category == "moss":
-                mod, params = moss.get_model(ARGS, config)
-            elif ARGS.model_category == "rwkv":
-                mod, params = rwkv.get_model(ARGS, config)
-            else:
-                raise ValueError(f"Model {ARGS.model} not supported")
-            mod = mod_transform_before_build(mod, params, ARGS)
-            with open(cache_path, "wb") as outfile:
-                pickle.dump(mod, outfile)
-            print(f"Save a cached module to {cache_path}.")
-            utils.copy_tokenizer(ARGS)
-        else:
-            print(
-                f"Load cached module from {cache_path} and skip tracing. "
-                "You can use --use-cache=0 to retrace"
-            )
-            with open(cache_path, "rb") as pkl:
-                mod = pickle.load(pkl)
-        dump_split_tir(mod, ARGS)
-        if not ARGS.reuse_lib:
-            build(mod, ARGS)
+    if ARGS.sep_embed and ARGS.model_category != "llama":
+        raise ValueError(f"separate embedding not supported on {ARGS.model}")
+    if ARGS.model_category != "minigpt":
+        with open(
+            os.path.join(ARGS.model_path, "config.json"), encoding="utf-8"
+        ) as i_f:
+            config = json.load(i_f)
+    if not use_cache:
+        if ARGS.model_category == "llama":
+            mod, params = llama.get_model(ARGS, config)
+        elif ARGS.model_category == "gpt_neox":
+            mod, params = gpt_neox.get_model(ARGS, config)
+        elif ARGS.model_category == "gpt_bigcode":
+            mod, params = gpt_bigcode.get_model(ARGS, config)
+        elif ARGS.model_category == "minigpt":
+            mod, params = minigpt.get_model(ARGS)
+        elif ARGS.model_category == "moss":
+            mod, params = moss.get_model(ARGS, config)
+        elif ARGS.model_category == "rwkv":
+            mod, params = rwkv.get_model(ARGS, config)
         else:
-            print("Reuse existing prebuilt lib {ARGS.reuse_lib}...")
-        dump_default_mlc_chat_config(ARGS)
+            raise ValueError(f"Model {ARGS.model} not supported")
+        mod = mod_transform_before_build(mod, params, ARGS)
+        with open(cache_path, "wb") as outfile:
+            pickle.dump(mod, outfile)
+        print(f"Save a cached module to {cache_path}.")
+        if ARGS.model_category != "minigpt":
+            utils.copy_tokenizer(ARGS)
```

Comment:
Because when building minigpt, we only build the vision module, and we don't need tokenizer for it since it's not a LLM module. The LLM module would be vicuna which is built and loaded separately.